src/main.go

```diff
@@ -187,13 +220,26 @@ func mainPage(c echo.Context) error {
 	if err != nil {
 		return c.Redirect(http.StatusFound, "/")
 	}
-	refreshChan := make(chan bool)
-	db := RROnline.Connect()
+	refreshChan := make(chan *RROnline.DBError)
+	db, dbErr := RROnline.Connect()
+	// TODO: Same deal as two functions up, should return a 5XX status code or render a special page
+	if err != nil {
+		RROnline.LogDBError(dbErr)
+	}
 	go RROnline.RefreshDevices(userToken.Value, db, refreshChan)
 	name := RROnline.GetUserName(userToken.Value)
 	email := RROnline.GetEmail(userToken.Value)
-	searches := RROnline.GetSearches(email, db)
-	interval := RROnline.GetInterval(email)
+	searches, dbErr := RROnline.GetSearches(email, db)
+	// TODO: Not sure what dbErr is, I think it changes? Either way if it being present undermines the following
```

Comment:
Unsure of the fatality of dbErr, but depending on that, this might need to end the process with a 5XX code or page rendering